src/crab/splitdbm/graph_ops.hpp

```
@@ -0,0 +1,659 @@
+            }
+        }
+    }
+}
+
+// ============================================================================
+// Internal helpers (used by the public scratch-based algorithms below)
+// ============================================================================
+namespace detail {
+
+// Compute the strongly connected components (Tarjan's algorithm).
+void strong_connect(ScratchSpace& scratch, const ReadableGraph auto& x, std::vector<VertId>& stack, int& index,
```

Clang (14.0.6)

[warning] parameter name 'x' is too short, expected at least 3 characters

(readability-identifier-length)